accepted/2024/[TS]PMUI-Readme-rendering.md

```diff
@@ -0,0 +1,116 @@
+
+We can use the Preview builder as follows:
+```C#
+//This creates a new instance of the preview builder
+var markdownPreview = new PreviewBuilder().Build();
+
+//We update the current markdown being rendered by calling "UpdateContentAsync"
+markdownPreview.UpdateContentAsync(markDown ?? string.Empty, ScrollHint.None)
+
+//IMarkdownPreview.VisualElement contains the FrameworkElement to be passed to the view
+MarkdownPreviewControl = markdownPreview.VisualElement
+```
+##### Locating the README
+We will check which options have been implemented by the feed for downloading the README file. 
+
+If the package metadata version indicates the RawReadmeFileUrl field should be available, then this will be the only option we use for downloading the ReadMe. 
+
+If the README resource has been added to the service index then we will use the URL format defined to download the README.
+This will involve checking that URL for every package. 
+
+If neither of these options are available we will use look for the README in the local file system.
+Meaning we will only render the README for already downloaded packages.
+
+It will be up to the individual feeds to implement a solution that works best for them, or even implement both. 
+
+## Drawbacks
+MarkdownPreview control currently marked as obsolete since the interface has not been finalized.
+So when an upgrade is made we may have to change how we use the control.
+
+WebView2 controls always render ontop of other controls in the view.
+[Secnario 25254665](https://microsoft.visualstudio.com/Edge/_workitems/edit/25254665).
+PM UI needs to be updated to ensure items don't scroll off screen.
+
+## Rationale and alternatives
+By using an existing control we maintain consistency throughout the IDE and can rely on the owner to fix any bugs with the control.
+
+Due to concerns about performance we will not be downloading the full nupkg temporarily just to access the README.
+
+## Prior Art
+The IMarkdownPreview is currently being used when creating a new pull request inside of Visual Studio.
+![Alt text](../../meta/resources/READMEPMUI/PullRequestExperience.png) 
+
+nuget.org currently renders the README and our users will expect them to look the same.
+Ex. https://www.nuget.org/packages/Newtonsoft.Json#README-body-tab
+
+## Unresolved Questions
+1. Will we show README in the updates tab?
+1. ~~What do we show if the package has a README.txt instead of md?~~
+    * The README is spec is written as only accepting MD.
+    So we will use the nuspec to determine where the README is and treat it as md, even if the file is actually txt. 
+1. ~~What do we show if there is no README defined?~~
+    * Show the README tab with a message saying there is no README for the selected pacakge/version.
```

Comment:
When you first open the package details view and a package has no readme, what would be the default? I'm assuming details since it's more usable?

Reply:
Right now the default is to whatever was used last. If a tab has never been selected it goes to the README tab.

Reply:
Should we go to the more useful tab for the package instead?